src/main/frontend/app/components/inputs/validatedInput.tsx

```diff
@@ -21,31 +21,34 @@ export default function ValidatedInput({
   const [checks, setChecks] = useState<Record<string, boolean>>({})
   const [inputValue, setInputValue] = useState(value)
 
+  const validateInput = useCallback(
+    (inputValue: string) => {
+      if (!patterns) {
+        setIsValid(true)
+        return
+      }
+
+      const newChecks: Record<string, boolean> = {}
+      for (const [name, pattern] of Object.entries(patterns)) {
+        newChecks[name] = pattern.test(inputValue)
+      }
+      const newIsValid = Object.keys(newChecks).length === 0 || Object.values(newChecks).every(Boolean)
+
+      setChecks(newChecks)
+      setIsValid(newIsValid)
+    },
+    [patterns],
+  )
```

Comment:
Again useCallback is used but no value seems to be returned so what is the purpose of memoising this function?